src/algebra/magma_algebra.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2021 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import algebra.algebra.basic
+import linear_algebra.finsupp
+
+/-!
+# Magma algebras
+
+If apply the construction of the group algebra but to a magma rather than a group, we obtain an
+algebra with a bilinear product, the so-called magma algebra.
+
+Although we cannot say much about such algebras since they sit in a very general class, they are
+useful for this same reason. In particular they can be used to construct free functors for
+categories of non-associative algebras, e.g., free Lie algebras.
+
+## Main definitions
+
+  * `magma_algebra`
+  * `foo`
```

Comment:
quux

Reply:
:-)

I've got a more sensible local version which I hope to complete tomorrow; I'll push it up now to remove this silliness.